cli/src/commands/config.rs

```diff
@@ -304,3 +317,17 @@ pub(crate) fn cmd_config_path(
     )?;
     Ok(())
 }
+
+fn format_origin(origin: Option<&str>, source: &ConfigSource) -> String {
+    let Some(origin) = origin else {
+        return format!("{source}:\t");
+    };
+
+    // `config::FileSourceFile::resolve()` returns a relative path. Try to
+    // convert them to absolute paths for easier recognition, falling back
+    // to the original value on failure.
+    let canon = Path::new(origin).canonicalize();
+    let path = canon.as_deref().unwrap_or_else(|_| Path::new(origin));
+
+    format!("{source}:{}\t", path.display())
```

Comment:
nit: use space instead of tab? I don't think `\t` is used elsewhere.

Reply:
Done as part of https://github.com/martinvonz/jj/pull/3023#discussion_r1485751209